### ISSUE TITLE
npx rnn-link is outdates aswell as documentation

### ISSUE THREAD

**Issue by rendomnet** [wix/react-native-navigation#7943]
### What happened?

npx rnn-link
is not working with android and kotlin files.

Documentation is also outdated
https://wix.github.io/react-native-navigation/next/docs/installing

### Was it tested on latest react-native-navigation?

- [X] I have tested this issue on the latest react-native-navigation release and it still reproduces.

### In what environment did this happen?

React Native Navigation version: "^7.40.3"
React Native version:  "0.76.3",   
Has Fabric (React Native's new rendering system) enabled: (yes/no) 
Node version: 18
Device model: 
Android version: 35

**Comment by retyui** [comment]
clone: https://github.com/wix/react-native-navigation/issues/7821

**Comment by retyui** [comment]
see https://github.com/wix/react-native-navigation/issues/7821#issuecomment-2514579985

**Comment by rendomnet** [comment]
Can somebody confirm that react-native-navigation is working with react-native 0.76.3?

**Comment by retyui** [comment]
@rendomnet Yes it's working. after linking you need to apply patches for android

do this setup https://www.npmjs.com/package/patch-package#set-up
and put this file in `./patches` [react-native-navigation+7.40.3.patch](https://github.com/user-attachments/files/18006034/react-native-navigation%2B7.40.3.patch) and run once `npx patch-package`

**Comment by rendomnet** [comment]
> @rendomnet Yes it's working. after [linking](https://github.com/wix/react-native-navigation/issues/7821#issuecomment-2514579985) you need to apply patches for android
> 
> do this setup https://www.npmjs.com/package/patch-package#set-up and put this file [react-native-navigation+7.40.3.patch](https://github.com/user-attachments/files/18006034/react-native-navigation%2B7.40.3.patch) into a `./patches` folder and run once `npx patch-package`

I tried it with blank new react-native project and it is not working for iOS and Android.
https://github.com/rendomnet/repro-bug-rnn

**Comment by rendomnet** [comment]
@retyui
This is what 'yarn ios' returns after completing your steps.

`error limit 19 -serialize-diagnostic-file /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-eqnrzhjzjreegcfjxgmvkdzfeljt/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/ReactNativeNavigation.build/Objects-normal/arm64/UIViewController+SideMenuController.dia -Wdeprecated-objc-isa-usage -Werror\=deprecated-objc-isa-usage -Werror\=implicit-function-declaration -Wnon-modular-include-in-framework-module -Werror\=non-modular-include-in-framework-module -Wno-trigraphs -Wno-missing-field-initializers -Wno-missing-prototypes -Werror\=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror\=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror\=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -Wno-implicit-fallthrough -Wprotocol -Wdeprecated-declarations -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -Wno-comma -Wno-shorten-64-to-32 -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -Wno-reserved-identifier -Wno-gnu-folding-constant -index-store-path /Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-eqnrzhjzjreegcfjxgmvkdzfeljt/Index.noindex/DataStore -index-unit-output-path /Pods.build/Debug-iphonesimulator/ReactNativeNavigation.build/Objects-normal/arm64/UIViewController+SideMenuController.o -fmodule-map-file\=/Users/user/Library/Developer/Xcode/DerivedData/AwesomeProject-eqnrzhjzjreegcfjxgmvkdzfeljt/`

**Comment by rendomnet** [comment]
@retyui 
and 'yarn android' is built but crashes immediately upon launch.

So my question still stands, can someone confirm that latest rnn is working with latest react-native. using the the guide above with newly created react-native blank project.

**Comment by gosha212** [comment]
iOS rnn link already implemented in #8037. Only android side left
